firmware/application/apps/pocsag_app.cpp

```diff
@@ -71,6 +71,8 @@ POCSAGAppView::POCSAGAppView(NavigationView& nav)
     if (!settings_.loaded())
         field_frequency.set_value(initial_target_frequency);
 
+    check_log.set_value(ui_settings_[enable_logging]->as_bool());
```

Comment:
I agree with you that this not very pretty.  Personally I think something like this would be more clear:
`    enable_logging = (bool)retrieve_saved_app_setting("enable_logging", DEFAULT_VALUE);`

Reply:
You still need to indicate the file to read from so free functions aren't a good choice. Also that would read/save on each call and saving ends up being a bunch of annoying IO to fix up the file. Finally, the radio settings need to also fit into this pattern and those need to all be loaded/saved before the UI controls have been constructed.

I'm starting to lean back toward my initial design where the settings are bound to pointers to the values they read/write. That way in the calling code you're just using variables instead of interacting with the settings object.

At first I thought it would be better to have the settings object actually hold the values, but the problem is that the SettingsVariant + its name was 36bytes per _instance_. The "binding" approach would be much more efficient than that.

Reply:
The binding approach would look like this in your class:
```
bool my_value;
SettingsStore("MyApp", {
    {"my_value_name", &my_value, false}
});
```
`my_value` would be read from the settings file on construction and then the value would be saved back on destruction. Does that seem better? I'm thinking it is.

Reply:
Yes that looks a lot better IMO.

Reply:
I've pushed another iteration that uses the "binding" approach.
I would appreciate if you took a quick look to double check. I have tested it out on my device and everything seems to be working fine.